src/towncrier/_builder.py

```diff
@@ -66,15 +66,15 @@ def parse_newfragment_basename(
 
 # Returns a structure like:
 #
-# OrderedDict([
+# {[
```

Comment:
The instantiation of OrderedDicts works a bit different than those of normal dicts.

What you've described here (and used in the tests) is creating a set with a list inside (which doesn't work).

The transformation is a bit tedious (you have to transform `[("foo", 42)]` into `{"foo": 42}`) so feel free to roll back that changes if you don't feel like it.